app/services/attribute_asserter.rb

```diff
@@ -59,6 +59,10 @@ def build
                 :decrypted_pii,
                 :user_session
 
+  def analytics
+    Analytics.new(user:, request: nil, session: {}, sp: nil)
+  end
```

Comment:
ditto about logging from the controller layer if possible! SAML idp controller is a mess but is there a way?

Reply:
this is actually way messier than the OIDC code path, since the `AttributeAsserter` object is used in private methods in the `SamlIdpAuthConcern` and is comingled with methods in the `SamlIdp` gem so i think it's not worth it here, especially for a temporary event.